src/MallardMessageHandlers/AuthenticationToken/AuthenticationTokenHandler.cs

```diff
@@ -25,8 +25,10 @@ namespace MallardMessageHandlers
 	public class AuthenticationTokenHandler<TAuthenticationToken> : DelegatingHandler
 		where TAuthenticationToken : IAuthenticationToken
 	{
+		private static readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1);
```

Comment:
We really need those variables static? oO

Reply:
yeah. especially if we want the semaphore to be shared for all authenticated endpoints. On MdC (at least), we have different instances of AuthenticationTokenHandler per Endpoints, since on project from UnoApplicationTemplate we declare our AuthenticationTokenHandler with AddTransient() https://github.com/nventive/UnoApplicationTemplate/blob/43b8ad861cad7fcd9dc392ef41e5e925c0e383d7/src/app/ApplicationTemplate.Shared/Configuration/ApiConfiguration.cs#L183.